kong/clustering/utils.lua

```diff
@@ -379,6 +414,17 @@ function _M.connect_cp(endpoint, conf, cert, cert_key, protocols)
     protocols = protocols,
   }
 
+  if conf.cluster_use_proxy then
+    local proxy_opts = parse_proxy_url(conf)
+    opts.proxy_opts = {
+      wss_proxy = proxy_opts.proxy_url,
+      wss_proxy_authorization = proxy_opts.proxy_authorization,
+    }
+
+    ngx_log(ngx_DEBUG, _log_prefix,
+            "using proxy ", proxy_opts.proxy_url, " to connect control plane")
```

Comment:
Will this URL contains the credentials?

Reply:
no it will be the scheme://host:port only